vcd/resource_vcd_lb_service_monitor.go

```diff
@@ -0,0 +1,339 @@
+package vcd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func ResourceVcdLbServiceMonitor() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdLbServiceMonitorCreate,
+		Read:   resourceVcdLbServiceMonitorRead,
+		Update: resourceVcdLbServiceMonitorUpdate,
+		Delete: resourceVcdLbServiceMonitorDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceVcdLbServiceMonitorImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"vdc": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"edge_gateway": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway name",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Unique Service Monitor name",
+			},
+			"interval": &schema.Schema{
+				Type:        schema.TypeInt,
+				Default:     10,
+				Optional:    true,
+				Description: "Interval at which a server is to be monitored",
+			},
+			"timeout": &schema.Schema{
+				Type:        schema.TypeInt,
+				Default:     15,
+				Optional:    true,
+				Description: "Maximum time in seconds within which a response from the server must be received",
+			},
+			"max_retries": &schema.Schema{
+				Type:        schema.TypeInt,
+				Default:     3,
+				Optional:    true,
+				Description: "Number of times the specified monitoring Method must fail sequentially before the server is declared down",
+			},
+			"type": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Way in which you want to send the health check request to the server",
+				ValidateFunc: validateLowerCase(),
+			},
+			"expected": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "String that the monitor expects to match in the status line of the HTTP or HTTPS response (for example, HTTP/1.1)",
+			},
+			"method": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Method to be used to detect server status",
```

Comment:
I was thinking of that but then I kind of refrained as the documentation states one list while the real API validator gives another one. Instead I made the `govcd` part so that errors are parsed and returned in nice format from NSX API which usually proposes possible options as well.

Reply:
I still recommend to write down possible values today - cause otherwise I have to find another documentation or guess and read errors when I run. I think current approach don't validation values though in terraform.
cc: @lvirbalas

Reply:
Ok. I've added at least some, but thinking to skip validation. If you miss the value - NSX API error is quite nicely formatted:
```
vcd_lb_service_monitor.lb-service-monitor[0]: Modifying... [id=monitor-297]

Error: unable to update load balancer service monitor with ID monitor-297: error while updating load balancer service monitor : vShield Edge [LoadBalancer] Invalid monitor type asd. Valid
monitor types are: HTTP|HTTPS|TCP. (API error: 14557)
```

**P.S.** You will probably notice that this list contradicts to list in documentation which I used - https://docs.vmware.com/en/vCloud-Director/9.0/com.vmware.vcloud.tenantportal.doc/GUID-DE3EDF95-1A8E-4D2F-B7B1-1254E9420868.html

Reply:
Thank you. Anyways I think @lvirbalas thinks to avoid specific validation for these cases even they longer to catch. As API could change in future, thouth we still could no support that until make API version more flexible.

Reply:
To me NSX error here is very clear. However, I can't match *our* documentation to what it writes. We have:

> method - (Optional) For types http and https. Select http method to be used to detect server status

Two issues here:
- No "TCP"
- Values are lower case, while NSX error shows upper case. As such as a user, I'm confused which to use.